app/models/user.rb

```diff
@@ -1,4 +1,6 @@
 class User < ActiveRecord::Base
+  attr_accessor :asserted_attributes
```

Comment:
Correction: I updated the previous comment to make `attr_reader :user` private because we don't want this class to expose the original user.